mpi.h

```diff
@@ -443,8 +443,7 @@ extern double MPI_Wtime(void);
 /*
  * Additional interfaces needed for compiling E3SM with gcc-14
  */
-extern int MPI_Get_Version(int *mpi_vers, int *mpi_subvers);
-extern int MPI_Get_library_version(char *version, int *resultlen);
```

Comment:
The declaration of MPI_Get_library_version() is already present, so the duplicate declaration has been removed in this PR.